### PR TITLE
ngspice: Add option to build from HEAD.

### PR DESCRIPTION
**Pull request by jmaibaum**
- [X] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [X] Does your submission pass `brew audit --new-formula <formula>` (after doing `brew install <formula>`)?

---

Add option to build ngspice from HEAD.

### CODE REVIEW

File: Formula/ngspice.rb [hunk 1]
```diff
@@ -16,7 +17,18 @@ class Ngspice < Formula
 
   depends_on :x11 => :optional
 
+  if build.head?
+    depends_on "autoconf" => :build
+    depends_on "automake" => :build
+    depends_on "bison" => :build
+    depends_on "libtool" => :build
+  end
```
Comment (bfontaine) [inline Formula/ngspice.rb]: Use the following instead:
```rb
head do
  url "git://ngspice.git.sourceforge.net/gitroot/ngspice/ngspice"

  depends_on "autoconf" => :build
  depends_on "automake" => :build
  depends_on "bison" => :build
  depends_on "libtool" => :build
end
```

File: Formula/ngspice.rb [hunk 2]
```diff
@@ -16,7 +17,18 @@ class Ngspice < Formula
...
+
   def install
+    if build.head?
+      system "./autogen.sh"
+    end
```
Comment (bfontaine) [inline Formula/ngspice.rb]: You can inline this: `system "./autogen.sh" if build.head?`.

**Comment by jmaibaum** [comment]
@bfontaine: I included your suggestions.

**Comment by MikeMcQuaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request merged**